### PR TITLE
1stmodel

### PR DESCRIPTION
**Pull request by katyachemistry**

### CODE REVIEW

**Review by Evgenii-Iurin — CHANGES_REQUESTED** [review]
Looks good! I would like to ask you to rewrite it using the standard machine learning library, PyTorch. Please note that in the Dataset class, features should never be generated or created. Instead, the Dataset class should only handle data indexing and retrieval through the __getitem__ and __len__ methods, while feature extraction or any data processing should be managed in a separate function or module

File: backend/ML/LGBM_1_day.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+def roll_mean_features(dataframe, windows):
+    for window in windows:
+        dataframe['cnt_roll_mean_' + str(window)] = dataframe.groupby(["store_id", "item_id"])['cnt'].                                                           transform(
+            lambda x: x.shift(1).rolling(window=window, min_periods=10, win_type="triang").mean()) + random_noise(
+            dataframe)
+    return dataframe
+
+# Metrics
+def smape(preds, target):
```
[the remark this answers is not in the document]
Comment (katyachemistry) [reply backend/ML/LGBM_1_day.py:28]: in sklearn there is only mape :(

File: backend/ML/LGBM_1_day.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        return self.model.predict(X, num_iteration=self.model.best_iteration)
+
+    def save_model(self, file_name):
+        self.model.save_model(file_name, num_iteration=self.model.best_iteration)
+
+if __name__ = "__main__":
+    dataset = Dataset('merged_df.csv')
+    model = Model()
+    model.train(dataset.lgbtrain, dataset.lgbval)
+    model.save_model('trial.txt')
```
[the remark this answers is not in the document]
Comment (katyachemistry) [reply backend/ML/LGBM_1_day.py:143]: well these are different stories. chatgpt: ".ckpt files are designed for saving neural network states (their weights) which has no relation to boosting models. converting between these formats is neither possible nor useful"

File: backend/ML/LGBM_1_day.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        cols = [col for col in train.columns if col not in ['date', 'date_id', "year", target_column]]
+        X_train = train[cols]
+        Y_train = train[target_column]
+        X_val = val[cols]
+        Y_val = val[target_column]
+        self.lgbtrain, self.lgbval = self.get_Lgb_dataset(X_train, Y_train, X_val, Y_val)    
+
+
+# Model Class
+class Model:
```
[the remark this answers is not in the document]
Comment (katyachemistry) [reply backend/ML/LGBM_1_day.py:103]: okie

File: backend/ML/LGBM_1_day.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Evgenii-Iurin) [reply backend/ML/LGBM_1_day.py:143]: Ok, can you provide the class method to load weights to model?

[the remark this answers is not in the document]
Comment (katyachemistry) [reply backend/ML/LGBM_1_day.py:143]: i will add method for loading a trained model

**Pull request merged**